### PR TITLE
Fix/401 redirects

### PR DESCRIPTION
**Pull request by svevang**
Fixes:

- Ensure that 401's from service call don't turn into 500s (that current_user is not truthy on redirect)
- Make sure that we don't lose our redirect destination on token expiration / back button.

### CODE REVIEW

File: lib/prx_auth/rails/ext/controller.rb [hunk 1]
```diff
@@ -8,19 +8,29 @@ class SessionTokenExpiredError < RuntimeError; end
...
       PRX_USER_INFO_SESSION_KEY = 'prx.auth.info'.freeze
       PRX_REFRESH_BACK_KEY = 'prx.auth.back'.freeze
 
       def prx_auth_token
-        env_token || session_token
-      rescue SessionTokenExpiredError
-        session.delete(PRX_JWT_SESSION_KEY)
-        session.delete(PRX_ACCOUNT_MAPPING_SESSION_KEY)
-        session.delete(PRX_USER_INFO_SESSION_KEY)
+        set_after_sign_in_path
```
Comment (cavis) [inline lib/prx_auth/rails/ext/controller.rb]: So we call this both before every action, and here whenever anyone asks for the `prx_auth_token` - why are both necessary?

File: lib/prx_auth/rails/ext/controller.rb [hunk 2]
```diff
@@ -8,19 +8,29 @@ class SessionTokenExpiredError < RuntimeError; end
...
+          session.delete(PRX_USER_INFO_SESSION_KEY)
+          nil
+        end
+      end
+
+      def set_after_sign_in_path
+        return if self.class == PrxAuth::Rails::SessionsController
+
         session[PRX_REFRESH_BACK_KEY] = request.fullpath
-        redirect_to PrxAuth::Rails::Engine.routes.url_helpers.new_sessions_path
```
Comment (cavis) [inline lib/prx_auth/rails/ext/controller.rb:23]: This seems like `SessionTokenExpiredError` will no longer trigger a redirect.  What does that impact?  (I'm trying to remember why that redirect was even necessary when rescuing the error).

Comment (svevang) [reply lib/prx_auth/rails/ext/controller.rb:23]: It's partially related to that `if current_user` containing the block that throws the 401 from the RPC to cms. The current user was getting back a redirect (truthy) that was then letting program flow to get down into the portion of the template containing the offending code.

This should have no practical effect, anyways we rely on `authenticate!` in the controllers to do this redirect.

**Review by cavis — APPROVED** [review]
👍 looks good to me!

**Pull request merged**